### PR TITLE
Added Edit Survey Functionality

### PR DESCRIPTION
**Pull request by mack-wallace**
Added an Edit survey controller along with HTML.

### CODE REVIEW

File: src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    SurveyRepository surveyRepository;
+    QuestionRepository questionRepository;
+    @Autowired
+    public WebEditSurveyController(SurveyRepository surveyRepository, QuestionRepository questionRepository){
+        this.surveyRepository = surveyRepository;
+        this.questionRepository = questionRepository;
+
+    }
+
+    @GetMapping(path = "survey/{surveyId}/edit")
```
Comment (DavidHaighton) [inline src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java:29]: based

**Review by DavidHaighton — COMMENTED** [review]
_                        _
| |__   __ _ ___  ___  __| |
| '_ \ / _` / __|/ _ \/ _` |
| |_) | (_| \__ |  __| (_| |
|_.__/ \__,_|___/\___|\__,_|

File: src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+@Log
+@Controller
+public class WebEditSurveyController {
+
+    SurveyRepository surveyRepository;
+    QuestionRepository questionRepository;
```
Comment (DavidSciola) [inline src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java:21]: Maybe make surveyRepository and questionRepository private.

File: src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @PostMapping("survey/rename")
+    public String editName(@RequestParam String prompt, @RequestParam long id) {
+        var survey = getSurvey(id);
+        survey.setName(prompt);
+        surveyRepository.save(survey);
+        return "redirect:/survey/"+id +"/edit";
+    }
+
```
Comment (DavidSciola) [inline src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java:65]: getSurvey and getQuestion are also defined in WebQuestionController, maybe make some helper class that has these methods.

File: src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java (hunk 2 above)
Comment (mack-wallace) [reply src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java]: done! :)

File: src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java (hunk 3 above)
Comment (mack-wallace) [reply src/main/java/com/based/basedsurvey/controller/WebEditSurveyController.java:65]: this can be done for milestone 2 #33

**Review by DavidSciola — APPROVED** [review]

**Pull request merged**